asdf_astropy/tags/transform/basic.py

```diff
@@ -0,0 +1,216 @@
+        model._initialize_constraints(param_and_model_constraints)
+
+        yield model
+
+        if 'inverse' in node:
+            model.inverse = node['inverse']
+
+    def from_yaml_tree(self, node):
+        raise NotImplementedError(
+            "Must be implemented in TransformType subclasses")
+
+    def from_tree(self, node):
+        model = self.from_yaml_tree(node)
+        return self._from_tree_base_transform_members(model, node)
+
+    def _to_tree_base_transform_members(self, model, node):
+        if getattr(model, '_user_inverse', None) is not None:
+            node['inverse'] = model._user_inverse
+
+        if model.name is not None:
+            node['name'] = model.name
+
+        try:
+            bb = model.bounding_box
+        except NotImplementedError:
+            bb = None
+
+        if bb is not None:
+            if model.n_inputs == 1:
+                bb = list(bb)
+            else:
+                bb = [list(item) for item in model.bounding_box]
+            node['bounding_box'] = bb
+        if type(model.__class__.inputs) != property:
+            node['inputs'] = model.inputs
+            node['outputs'] = model.outputs
+
+        # model / parameter constraints
+        if not isinstance(model, CompoundModel):
+            fixed_nondefaults = {k: f for k, f in model.fixed.items() if f}
+            if fixed_nondefaults:
+                node['fixed'] = fixed_nondefaults
+            bounds_nondefaults = {k: b for k, b in model.bounds.items() if any(b)}
+            if bounds_nondefaults:
+                node['bounds'] = bounds_nondefaults
+
+        return node
+
+    def to_yaml_tree(self, model):
+        raise NotImplementedError("Must be implemented in TransformType subclasses")
+
+    def to_tree(self, model):
+        node = self.to_tree_transform(model)
+        return self._to_tree_base_transform_members(model, node)
+
+    def assert_equal(self, a, b):
+        # TODO: If models become comparable themselves, remove this.
+        assert a.name == b.name
+        # TODO: Assert inverses are the same
+
+
+class IdentityType(TransformType):
+    tags = {"http://asdf-format.org/schemas/transform/identity-2.0.0"}
+    types = {models.Identity}
+
+    def from_yaml_tree(self, node):
+        return models.Identity(node.get('n_dims', 1))
+
+    def to_yaml_tree(self, model):
+        node = {}
+        if model.n_inputs != 1:
+            node['n_dims'] = model.n_inputs
+        return node
+
+    def assert_equal(self, a, b):
+        # TODO: If models become comparable themselves, remove this.
+        TransformType.assert_equal(a, b)
+        assert (isinstance(a, models.Identity) and
+                isinstance(b, models.Identity) and
+                a.n_inputs == b.n_inputs)
+
+
+class ConstantType(TransformType):
+    tags = {"http://asdf-format.org/schemas/transform/constant-2.0.0"}
+    supported_versions = ['1.0.0', '1.1.0', '1.2.0', '1.3.0', '1.4.0', '2.0.0']
```

Comment:
`supported_versions` is also now ignored, that is determined by the list of tags.